src/error.rs

```diff
@@ -0,0 +1,24 @@
+use reqwest::{header::InvalidHeaderValue, StatusCode};
+use thiserror::Error;
+
+use crate::responses;
+
+#[derive(Error, Debug)]
+pub enum Error<Respone> {
+    #[error("encountered an error when performing an HTTP request")]
+    RequestError(#[from] reqwest::Error),
+    #[error("API responded with a client error: status code of {0}")]
+    ClientErrorResponse(StatusCode, Respone),
+    #[error("API responded with a server error: status code of {0}")]
+    ServerErrorResponse(StatusCode, Respone),
+    #[error("Health check failed: resource alarms are in effect")]
+    HealthCheckFailed(responses::HealthCheckFailureDetails),
+    #[error("Could not find the requested resource")]
+    NotFound(),
+    #[error("Can't delete a binding: multiple matching bindings found")]
```

Comment:
I've been contributing to RabbitMQ and/or clients libraries since 2010 and I have never seen this error. But it makes some sense, the only way to differentiate between multiple bindings between a source exchange and its destination is via their properties and they were not provided, returning an error is an option.